common/models/templates/test/test-tokenization.js

```diff
@@ -0,0 +1,358 @@
+      let expectedResult = {
+        left: ['The', 'quick', 'brown', 'fox', ''],
+        right: ['jumped', 'over', 'the', 'lazy', 'dog'],
+        caretSplitsToken: true
+      };
+
+      assert.deepEqual(tokenization, expectedResult);
+    });
+
+    it('tokenizes English using defaults, post-whitespace caret, partial context', function() {
+      let context = {
+        left: "quick brown fox ",
+        right: "jumped over the lazy",
+        startOfBuffer: false,
+        endOfBuffer: false
+      };
+
+      let tokenization = models.tokenize(wordBreakers.default, context);
+
+      // Technically, we're editing the start of the first token on the right
+      // when in this context.
+      let expectedResult = {
+        left: ['quick', 'brown', 'fox', ''],
+        right: ['jumped', 'over', 'the', 'lazy'],
+        caretSplitsToken: true
+      };
+
+      assert.deepEqual(tokenization, expectedResult);
+    });
+
+    it('tokenizes English using defaults, splitting caret', function() {
+      let context = {
+        left: "The quick brown fox jum",
+        right: "ped over the lazy dog",
+        startOfBuffer: true,
+        endOfBuffer: true
+      };
+
+      let tokenization = models.tokenize(wordBreakers.default, context);
+
+      let expectedResult = {
+        left: ['The', 'quick', 'brown', 'fox', 'jum'],
+        right: ['ped', 'over', 'the', 'lazy', 'dog'],
+        caretSplitsToken: true
+      };
+
+      assert.deepEqual(tokenization, expectedResult);
+    });
+
+    it('empty context case', function() {
+      // Wordbreaking on a empty space => no word.
+      let context = {
+        left: '', startOfBuffer: true,
+        right: '', endOfBuffer: true
+      };
+  
+      let tokenization = models.tokenize(wordBreakers.default, context);
+
+      let expectedResult = {
+        left: [],
+        right: [],
+        caretSplitsToken: false
+      };
+  
+      assert.deepEqual(tokenization, expectedResult);
+    });
+
+    it('nil context case', function() {
+      // Wordbreaking on a empty space => no word.
+      let tokenization = models.tokenize(wordBreakers.default, null);
+
+      let expectedResult = {
+        left: [],
+        right: [],
+        caretSplitsToken: false
+      };
+  
+      assert.deepEqual(tokenization, expectedResult);
+    });
+
+    it('near-empty context:  one space before caret', function() {
+      // Wordbreaking on a empty space => no word.
+      let context = {
+        left: ' ', startOfBuffer: true,
+        right: '', endOfBuffer: true
+      };
+  
+      let tokenization = models.tokenize(wordBreakers.default, context);
+
+      let expectedResult = {
+        left: [''],
+        right: [],
+        caretSplitsToken: false
+      };
+  
+      assert.deepEqual(tokenization, expectedResult);
+    });
+
+    // For the next few tests:  a mocked wordbreaker for Khmer, a language
+    // without whitespace between words.
+    let mockedKhmerBreaker = function(text) {
```

Comment:
New test-feature:  tests model tokenization for languages that don't use whitespace between words!

To be fair, this 'wordbreaking' is mocked... but still, that gives us a major leg up for proper use of such wordbreakers in the future.

Reply:
I'm really happy to see these Khmer tests!